v94.3.0

fluxminds opened the pull request:
# 94.3.0

This minor release primarily consists of accessibility improvements. A number of components now provide increased support for VoiceOver and Dynamic Type, and many components have been audited to make sure minimum touch target requirements and minimum color contrast requirements are met. Additionally, many examples in our Catalog and Dragons apps have been improved from an accessibility standpoint.

## New deprecations

* `adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable` in `MDCSnackbarManager`

* `adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable` in `MDCSnackbarMessageView`

## Component changes

### ActionSheet

* [Allow Swift example to support dynamic type (#8946)](https://github.com/material-components/material-components-ios/commit/5310c5c448776beec0f5b379b47b10c497134a9c) (Cody Weaver)

### ActivityIndicator

* [Add a voiceover layout change notification… (#8909)](https://github.com/material-components/material-components-ios/commit/c0a0d2b1de326006ac35a133559f28ea04a0b722) (Yarden Eitan)

### AppBar

* [AppBar example has low-contrast text (#8966)](https://github.com/material-components/material-components-ios/commit/af583a58285bc2ddb3b70d7a0d1d880ff25133e7) (Galia Kaufman)
* [Make example titles more accurate. (#8828)](https://github.com/material-components/material-components-ios/commit/bd832a416caa863dea3bd283164c93fd0d90ac44) (Robert Moore)

### Banner

* [Add a snapshot for banner with no action case. (#8968)](https://github.com/material-components/material-components-ios/commit/e1ceaf955b7a3f4ee0f94d16dae1a1e10e025741) (Wenyu Zhang)
* [Fix Banner positioning in example for iOS 10 (#8829)](https://github.com/material-components/material-components-ios/commit/bd0d571ae35eb54b29c0e6bf4cbf32c40c593753) (Bryan Oltman)
* [Support use case where no button appears on Banner. (#8951)](https://github.com/material-components/material-components-ios/commit/97e435ae881fe5e755196fcac6951f4a7b6f5c76) (Wenyu Zhang)

### BottomNavigation

* [Add accessibility value for badge (#8920)](https://github.com/material-components/material-components-ios/commit/ba3bc2b15a3376f412e12d8f00590fdec7b9a354) (Cody Weaver)
* [Clean up badge value test example (#8921)](https://github.com/material-components/material-components-ios/commit/98ec8abdbfdc9ffbd8a817c5f18030ebdd3cda80) (Cody Weaver)

### BottomSheet

* [Add VoiceOver support to BottomSheetShortCollectionExample (#8849)](https://github.com/material-components/material-components-ios/commit/adfe99ed30c5a46eb264b692e1cbf2b690e701fe) (Bryan Oltman)
* [Add accessibility elements to BottomSheetExamples (#8923)](https://github.com/material-components/material-components-ios/commit/78fe3701d073eb1fe932df507d0205da0e0e2ed2) (Bryan Oltman)
* [Make bottom sheet be in extended state when voiceover is on (#8787)](https://github.com/material-components/material-components-ios/commit/7807cf53fa10a1af71501d9509eb485ddd5b7b09) (Andrew Overton)
* [Remove scrim accessibility logic from BottomSheetUIControlExample (#8848)](https://github.com/material-components/material-components-ios/commit/5b826f163a574291f7193450907988e687a69f60) (Bryan Oltman)

### ButtonBar

* [Darken ButtonBarIconExample icon colors (#8826)](https://github.com/material-components/material-components-ios/commit/4baa770fde0c12dcd9583fb91a24bc3c98d1061b) (Bryan Oltman)
* [Delete the Theming extension. (#8696)](https://github.com/material-components/material-components-ios/commit/fb36f304ba1ed53608606763be45dee2bc47d15e) (featherless)

### Buttons

* [Fix a11y issues in Buttons (Content Edge Insets) exam… (#8900)](https://github.com/material-components/material-components-ios/commit/f6855caa775a461ca0cc1b3f433cfd250b2a465e) (Yarden Eitan)
* [Improve accessibility of Shaped Buttons. (#8964)](https://github.com/material-components/material-components-ios/commit/b12a4cba54b0c11a93da4912d521b4984000c6c2) (Robert Moore)
* [Update accessibility label for FABs (#8911)](https://github.com/material-components/material-components-ios/commit/98cafa6756916de88b7ac429d118364870f09fb0) (Cody Weaver)

### Cards

* [Fix bug in the "Typical use" example where the image would not appear. (#8943)](https://github.com/material-components/material-components-ios/commit/be33d3196a60602de1f367e1efef1f77bb096cb1) (featherless)
* [Make the "Typical use" example title label an a11y heading. (#8945)](https://github.com/material-components/material-components-ios/commit/f6f2ff28215ef572c0b3b384992ee2c873f2828a) (featherless)
* [Make the Shaped Edit/Reorder example class/files match the name of the example. (#8899)](https://github.com/material-components/material-components-ios/commit/ae7db6f03d55a69af118cbe5736d767355fd9a65) (featherless)
* [Make the shaped collection cell an accessibility button. (#8901)](https://github.com/material-components/material-components-ios/commit/28b838cc9da7bdc6dc9f2799140acb6bfa51923a) (featherless)

### Chips

* [Fix layout of ChipsSizingExampleViewController. (#8940)](https://github.com/material-components/material-components-ios/commit/e9e913b4dddcf57a4879bdc92723a42d06a86e63) (featherless)
* [Give the ChipsSizing example's sliders a11y labels and values. (#8942)](https://github.com/material-components/material-components-ios/commit/a0dd0a34b3b756fef038c6a62949e720d2096f93) (featherless)
* [Set UIAccessibilityTraitButton on MDCChipView's accessibilityTraits. (#8792)](https://github.com/material-components/material-components-ios/commit/edca5401bf810572a6fcab9ff0bd90acd73b4282) (featherless)

### Dialogs

* [Allow clients to customize message accessibility label  (#8801)](https://github.com/material-components/material-components-ios/commit/8becfc1ec55d2142d09b22700fd8980ab9c301e8) (Cody Weaver)
* [Allow clients to customize title accessibilityLabel. (#8799)](https://github.com/material-components/material-components-ios/commit/0796e543321a6e059b401a04c115ac588b101eb2) (Cody Weaver)
* [Fix bug in "Dismissing Dialogs" dialogs where the buttons were not themed. (#8967)](https://github.com/material-components/material-components-ios/commit/a331cb6a1a8dbd704fe5d86e63c00bdf50e86bc7) (featherless)
* [Implement the a11y escape gesture on all custom dialogs in the "Dismissing Dialogs" example. (#8965)](https://github.com/material-components/material-components-ios/commit/5d8e992963920b70c1ef609af0fb6593ba157daf) (featherless)
* [Make all "Dismissing Dialogs" cells a11y buttons. (#8956)](https://github.com/material-components/material-components-ios/commit/7cfb460a01341b67efa58a7a34e0937b92d0b605) (featherless)
* [Make the "Dialogs with accessory view" example's cells accessibility buttons. (#8939)](https://github.com/material-components/material-components-ios/commit/88d3bed28ab87a975ed74799b3a22941a68aa1c3) (featherless)
* [Make the "View with corner radius" example's dialog escapable in VoiceOver. (#8950)](https://github.com/material-components/material-components-ios/commit/34b0ad9ba249e125cd0ca74341d8e7dafdcbb841) (featherless)
* [More Material Alerts examples is more accessible. (#8960)](https://github.com/material-components/material-components-ios/commit/bde962e51179989977b819b92f3e2580c193b5f4) (Robert Moore)
* [Provide accessibility traits for Input Field example (#8958)](https://github.com/material-components/material-components-ios/commit/32ba0c927e6cc0a2ac1100a5e2a2f5a462d4c473) (Robert Moore)
* [Remove usage of `buttonFont` in example (#8772)](https://github.com/material-components/material-components-ios/commit/524ca5cab9874b437fef0c0138b3ee9f8bf99cb6) (Cody Weaver)
* [Set dialog message accessibilityFrame based on visible message text (#8786)](https://github.com/material-components/material-components-ios/commit/c83333f56518399f374b1b8914eac0d853fa20da) (Bryan Oltman)

### FeatureHighlight

* [Add a voiceover dismissal affordance for the feature highlight. (#8959)](https://github.com/material-components/material-components-ios/commit/4c7402857c173db101f846409ad716fac8ada4a3) (Yarden Eitan)
* [Make "Color" example accessible. (#8957)](https://github.com/material-components/material-components-ios/commit/8e9eff85f846f15e2bb954854586c91fe7bf814b) (Robert Moore)
* [Positions label not hidden behind nav bar. (#8969)](https://github.com/material-components/material-components-ios/commit/db709613cc58397dfb2ee7a9d0d21a8be8d9a2d9) (Yarden Eitan)

### Ink

* [Consolidate example code into a single file. (#8890)](https://github.com/material-components/material-components-ios/commit/cdaf0977567486546a633704809c759211f80b94) (featherless)
* [Make the ink views accessibility elements. (#8891)](https://github.com/material-components/material-components-ios/commit/baf74b9336ca8ac12422a7291bd85cad0acd0237) (featherless)

### List

* [Fix MDCBaseCell Example to work with Voice Over (#8953)](https://github.com/material-components/material-components-ios/commit/722399ddce71f214211651d01bdbf344f9c9829c) (Galia Kaufman)
* [Fix accessibility labels and layout in CollectionListCellExampleTypicalUse (#8944)](https://github.com/material-components/material-components-ios/commit/c89ac31d2b02bf345c97c0019ed132d63149def9) (Bryan Oltman)
* [Fix catalog compile issue (#8974)](https://github.com/material-components/material-components-ios/commit/f660c8a73226410927cb906155e76a198d53d0b8) (Cody Weaver)
* [Make Base Cell example more accessible. (#8947)](https://github.com/material-components/material-components-ios/commit/009af8be8159ff460bfa8908dd2d320eec8ad9a4) (Robert Moore)
* [Make Base Cell example more accessible. (#8982)](https://github.com/material-components/material-components-ios/commit/90fcc9281a2e8505bc8cd6daaf42efd4ce04f0ef) (Robert Moore)

### NavigationDrawer

* [Add close button to bottom drawer example header (#8963)](https://github.com/material-components/material-components-ios/commit/e67d2e5b660d1880bc879c4377cc231beb8b749b) (Bryan Oltman)
* [Remove container view when dismiss transitio… (#8660)](https://github.com/material-components/material-components-ios/commit/36d2ec7618fc8dcc3b4cf57759e3c84674eabee9) (Jonathan Willing)

### PageControl

* [Fixes contrast ratio and button interactivity for… (#8922)](https://github.com/material-components/material-components-ios/commit/e550e27089883cb4bde1de40341c5d1d9c1ed107) (Yarden Eitan)
* [Improve contrast ratio in PageControlSwiftExampleViewController (#8938)](https://github.com/material-components/material-components-ios/commit/111da2c00c44a704e7e812a9d31c05cae99bb4f0) (Bryan Oltman)
* [Improve contrast ratio in PageControlTypicalUseViewController (#8941)](https://github.com/material-components/material-components-ios/commit/f40e7f09ca6e05212563cad7eb862f7b7ec6a6f9) (Bryan Oltman)
* [adds a11y support for page control with next button example (#8932)](https://github.com/material-components/material-components-ios/commit/f58c3451f5dab1482448c2f7838559873dda39de) (Yarden Eitan)

### ProgressView

* [Add accessibilityLabel (#8775)](https://github.com/material-components/material-components-ios/commit/a049592f235a23b7f563f910e5fce786b350d7a5) (Bryan Oltman)

### Ripple

* [Add tests demonstrating unexpected layer.mask behavior. (#8808)](https://github.com/material-components/material-components-ios/commit/9123dc603e62d8068d5f3ed7423216148b1bfde1) (featherless)
* [Ripple typical example now can be used with Voice Over (#8924)](https://github.com/material-components/material-components-ios/commit/83d465eedc8dd8e7007a34448708b1ca0b45aaee) (Yarden Eitan)
* [make card with ripple example a5able (#8929)](https://github.com/material-components/material-components-ios/commit/b18c7481239a2cf4c6d9141caebbe9947426ef76) (Yarden Eitan)

### ShadowElevations

* [Delete non-functional ShadowElevationsTypicalUseViewController.swift (#8955)](https://github.com/material-components/material-components-ios/commit/655604fe58e82b9fb967e608d34c34984c111d8f) (Bryan Oltman)
* [Provide semantic VoiceOver labels when interacting with the typical use example. (#8930)](https://github.com/material-components/material-components-ios/commit/74f6530d154a58886f3b7ad2dd619628d370d807) (featherless)
* [Delete swift example target since there is no input file. (#8955)](https://github.com/material-components/material-components-ios/commit/ef4aad367748f60ddc4fd31cfaadf1a005cd76e7) (Wenyu Zhang)

### ShadowLayer

* [Make Shadow Corner Radius example more accessible. (#8962)](https://github.com/material-components/material-components-ios/commit/69e8021b2af5627534b5051ab832d74b02604157) (Cody Weaver)

### Slider

* [Update documentation to match behavior. (#8782)](https://github.com/material-components/material-components-ios/commit/3671e05bde32bf2bfb693d3eb6e21dcf87abe4e8) (Robert Moore)

### Snackbar

* [Add invisible hit target for legacy snackbar (#8807)](https://github.com/material-components/material-components-ios/commit/bd938100fb6a0b9fc000e16c6dbd113e4f558dc0) (Bryan Oltman)
* [Deprecate adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable (#8728)](https://github.com/material-components/material-components-ios/commit/6717fca5942b5fb49b0c3678746a5e63f90e6877) (Bryan Oltman)

### Tabs

* [Add newlines after all Swift doc headers. (#8931)](https://github.com/material-components/material-components-ios/commit/71ad8292e155152813dce8dd22456108e094c0b3) (featherless)
* [Improve accessibility for TabBar example. (#8983)](https://github.com/material-components/material-components-ios/commit/93700ab4d9ae04afb24cb45dead0585fed814e7c) (Robert Moore)
* [Improve custom view documentation. (#8813)](https://github.com/material-components/material-components-ios/commit/deab4b91ffae9aa04df921fd0b997dbaf39fbefa) (Robert Moore)
* [Mark TabBarView as accessibility tab bar. (#8811)](https://github.com/material-components/material-components-ios/commit/ed8acbbc71a919f7340b9b0e38295fd9d56bd903) (Robert Moore)
* [Use built-in "tabs" behavior on iOS 10+ (#8812)](https://github.com/material-components/material-components-ios/commit/a4a17a58dac1c6068f275e54b30961605bde808e) (Robert Moore)

### TextControls

* [Add adjusts content size category behavior (#8802)](https://github.com/material-components/material-components-ios/commit/c1dc83ed445b6dcab734251bf7d838b94020a0ff) (Andrew Overton)
* [Add dynamic type functionality to text controls example (#8847)](https://github.com/material-components/material-components-ios/commit/d93ddfbc1be6b65606fed710a077cb6bbd6fc348) (Andrew Overton)
* [Snapshot test with AXXXL sizes (#8781)](https://github.com/material-components/material-components-ios/commit/53a925d96d101777080aea05e5f1d9125fa12708) (Andrew Overton)
* [Use system dynamic colors for sensible defaults in iOS 13 (#8818)](https://github.com/material-components/material-components-ios/commit/9b9edf9f462fd0470a6a4486fd8e013471f69574) (Andrew Overton)

### TextFields

* [Added accessibility label tests for MDCMultilineTextFields (#8902)](https://github.com/material-components/material-components-ios/commit/1c17792b3656ff2f73f2261ddc9e2d1b0b1b78ed) (Randall Li)
* [Added header comment: placeholder APIs are for `label text` (#8803)](https://github.com/material-components/material-components-ios/commit/3c0d7953ada34365c0de9aed9664eed46984f431) (Randall Li)
* [Added tests for accessibilityLabel (#8830)](https://github.com/material-components/material-components-ios/commit/3d1b6f9acc0eeea2f5a2ae820d7c6ede437df628) (Randall Li)
* [Check if placeholder is hidden before verbalizing it in voice over. (#8804)](https://github.com/material-components/material-components-ios/commit/f9c17caf1bffd8e4538ee2c587d780e55437243c) (Randall Li)
* [Convert clear button to MDCButton (#8784)](https://github.com/material-components/material-components-ios/commit/42e220d9c1bcf133a1dcf1186023395a4303a0e9) (Bryan Oltman)
* [Correction to the accessibility label (#8976)](https://github.com/material-components/material-components-ios/commit/c498516b358998c9853e9934df9b643d4256b9da) (Randall Li)
* [Fix spelling mistake (#8910)](https://github.com/material-components/material-components-ios/commit/a4decc6c8e17ecab74aceab006bf36e676c3acfa) (Randall Li)
* [Update announcement text for fields with a character limit (#8777)](https://github.com/material-components/material-components-ios/commit/be03cec76744914a7861e9dc6d77c6fcd5e38bef) (Bryan Oltman)
* [Update character count announcement in MDCTextInputControllerFullWidth (#8789)](https://github.com/material-components/material-components-ios/commit/118dfa3d58b69fed2d8bbfdbaf263a4cbf51e1ca) (Bryan Oltman)
* [simplify test names since the name of the file already gives context. (#8952)](https://github.com/material-components/material-components-ios/commit/d6ae7f6782bb554d043879b5965e9b2708145a60) (Randall Li)
* [Add accessibilityLabel override to MDCBaseTextField (#8773)](https://github.com/material-components/material-components-ios/commit/d5e69512644fee50c3c78e44ef62242826b2785a) (Andrew Overton)

### Typography

* [Using safeSharedApplication in test (#8815)](https://github.com/material-components/material-components-ios/commit/afbd7351b50bcb8c0c7b97ec5e12f3002d6021c9) (Randall Li)

### schemes/Typography

* [Separate protocol target. (#8798)](https://github.com/material-components/material-components-ios/commit/692c743f34421d4bf06acc733f1f77886d65b092) (Robert Moore)

## Multi-component changes

* [Annotate all catalogShouldHideNavigation invocations with @objc. (#8791)](https://github.com/material-components/material-components-ios/commit/5425884431d6cd48d3ebb5f875d859a3a08ed059) (featherless)

fluxminds committed:
[Banner] update constraint when frame is changed. (#8765)

Because Banner needs to update constraints when there is a bound change, and when some view controller in the view hierarchy changes Banner's frame, it might forget to update constraint. Setting it in the frame setter would help making sure it is not missed.

bryanoltman committed:
[Snackbar] Deprecate adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable (#8728)

* [Snackbar] Deprecate adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable

* Revert change to default value

* More reverting changed to default behavior

* clang

codeman7 committed:
[Dialogs] Remove usage of `buttonFont` in example (#8772)

The `buttonFont` API is deprecated within MDCAlertController. This change removes it's usage from the catalog example. This removal brings us closer to being able to delete the API.

## Screenshots

| Before | After |
| --- | --- |
|![Simulator Screen Shot - iPhone 7 - 2019-11-11 at 10 31 29](https://user-images.githubusercontent.com/7131294/68611204-6ecc0700-046e-11ea-9b8b-fe6ca39f18ee.png)|![Simulator Screen Shot - iPhone 7 - 2019-11-11 at 10 22 36](https://user-images.githubusercontent.com/7131294/68611214-72f82480-046e-11ea-8b26-50c693d791c1.png)|



Related to #6637

bryanoltman committed:
[ProgressView] Add accessibilityLabel (#8775)

Commit:
[Slider][ThumbTrack] Fix event handling. (#8759)

Slider/ThumbTrack event handling was broken by the way track tick visibility
was implemented. This PR fixes that event handling.

Fixes #8758

Commit:
[Slider] Update documentation to match behavior. (#8782)

The MDCSlider documentation needs to be updated to match its current behavior.
This commit corrects an outdated behavioral description for the
`trackTickVisibility` API. It also updates the README to include more recent
changes to the APIs.

Follow-up from #8758

andrewoverton committed:
[TextControls] Add accessibilityLabel override to MDCBaseTextField (#8773)

UITextField has some VoiceOver support built in--the text the user enters is reflected in the accessibilityValue. The accessibilityLabel, however, seems to be nil unless a user specifies something for it. This PR doesn't change that behavior, but it does make it so that if no accessibilityLabel is set the textfield returns a concatenated string consisting of the label text and any text from the assistive labels.

Closes #8771.

fluxminds committed:
[AppBar] Add API to auto adjust height based on HeaderStackView for app bar. (#8764)

Part of https://github.com/material-components/material-components-ios/issues/8677.

This API helps to reduce the complexity of calculating minimumHeight and maximumHeight on FlexibleHeader when user wants to update the app bar height based on header stack view.

This flag is required to be set during configuration stage. The reset behavior (turning back to NO) was not implemented because it would require adding API to minMaxHeight class and all configuration needs be to be set before flexible header is used. This feature can be added if needed.

bryanoltman committed:
[Dialogs] Set dialog message accessibilityFrame based on visible message text (#8786)

bryanoltman committed:
[TextFields] Update announcement text for fields with a character limit (#8777)

Commit:
[ButtonBar] Delete the Theming extension. (#8696)

This target is deprecated and unused internally.

Part of #8429

andrewoverton committed:
[BottomSheet] Make bottom sheet be in extended state when voiceover is on (#8787)

This PR makes it so that the MDCBottomSheetController's MDCSheetState property accurately reflects the actual state its in during VoiceOver. Its previous VoiceOver behavior made it _look_ like it was extended, but `po bottomSheetController.state` would yield `MDCSheetStatePreferred`. Now it outputs `MDCSheetStateExtended`.

Thoroughly manually tested.

Closes #6758.

Commit:
[examples] Annotate all catalogShouldHideNavigation invocations with @objc. (#8791)

Context:

Due to changes in the Swift runtime over the past years, methods that are considered "optional" in Objective-C need to have @objc annotations in order for Objective-C to be able to perform `respondsToSelector:` invocations.

Problem:
Many of our examples had not annotated their `catalogShouldHideNavigation` methods with the `@objc` annotation, causing the Catalog to wrap the examples in a container app bar view controller. The reason most of these examples would implement `catalogShouldHideNavigation` is because they are managing their own navigation (e.g. to demonstrate AppBar functionality), so this would often result in duplicate AppBars appearing in the example.

The fix:
All `catalogShouldHideNavigation` Swift methods have been annotated with `@objc`.

Fixes https://github.com/material-components/material-components-ios/issues/7576
Fixes https://github.com/material-components/material-components-ios/issues/7578

bryanoltman committed:
[TextFields] Update character count announcement in MDCTextInputControllerFullWidth (#8789)

fluxminds committed:
[Banner] Deactivate custom constraints when there is a need to update constraints.(#8793)

Commit:
[Chips] Set UIAccessibilityTraitButton on MDCChipView's accessibilityTraits. (#8792)

Fixes https://github.com/material-components/material-components-ios/issues/8788

fluxminds committed:
[Banner] Add an example to show how Banner can be used as bottom bar in AppBar. (#8785)

Closes https://github.com/material-components/material-components-ios/issues/8677.

This PR adds an example to show how Banner could be used as a bottomBar in AppBar.

|Screenshot|
|-----|
|![appbarbanner](https://user-images.githubusercontent.com/8836258/68698933-005f7580-0550-11ea-957f-d8121e3f468d.gif)|

Commit:
Replace featherless with jverkoey in CODEOWNERS (#8797)

@jverkoey is my GitHub handle. @featherless is my personal org.

bryanoltman committed:
[TextFields] Convert clear button to MDCButton (#8784)

* [TextFields] Convert clear button to MDCButton

* Add buttons to build file

* Build file update

* Update build file again

* [Dialogs] Set dialog message accessibilityFrame based on visible message text

* Revert "[Dialogs] Set dialog message accessibilityFrame based on visible message text"

This reverts commit ae6154ed8b2df5a3ad73ad2e7923ed404297f821.

* Add snapshot test for clear button ripple

* Add ripple snapshot goldens

* Remove unnecessary cornerRadius setting

* Cast clear button ink color alpha to CGFloat

* Add Button and Ripple dependencies to snapshot tests

* PR feedback

* clang

* Fix imports

jwilling committed:
[NavigationDrawer] Remove container view when dismiss transitio… (#8660)

For cases where a sheet is reused, a new container view would be created without the existing view being removed.

Commit:
[TypographyScheme] Separate protocol target. (#8798)

Splits the `MDCTypographyScheming` protocol target from the implementation
target. This allows clients to conform to the protocol without depending on
the MDC-iOS implementation (and its dependencies).

Fixes #8445

andrewoverton committed:
Snapshot test with AXXXL sizes (#8781)

Note: This PR no longer relates to the automatic adjusting behavior on MDCBaseTextField. Instead, it is solely about adding snapshot test coverage for accessibility XXXL content sizes.

Closes #8800.

codeman7 committed:
[Dialogs] Allow clients to customize title accessibilityLabel. (#8799)

This change allows clients to customize the title accessibility label.

Related to #8779

galiak11 committed:
merged release-candidate

codeman7 committed:
[Dialogs] Allow clients to customize message accessibility label  (#8801)

This change allows clients to customize the message accessibility label.

Closes to #8779

Commit:
[Ripple] Add tests demonstrating unexpected layer.mask behavior. (#8808)

These tests demonstrate how a ripple view added as a subview can end up with an incorrect layer mask if the ripple view's frame does not match the bounds of the parent view and the parent view's layer has a shadowPath. All of the tests intentionally pass, with the final test demonstrating the incorrect behavior.

In a follow-up change, I will be adding a behavioral flag to turn off this unexpected behavior in favor of a more explicit approach to configuring the ripple view's layer mask.

Part of https://github.com/material-components/material-components-ios/issues/6913

andrewoverton committed:
[TextControls] Add adjusts content size category behavior (#8802)

This PR adds `UIContentSizeCategoryDidChangeNotification` observing and an implementation of `UIContentSizeCategoryAdjusting`.

Snapshot tests validating behavior in larger content sizes were added in #8781.

Closes #8770.

bryanoltman committed:
[Snackbar] Add invisible hit target for legacy snackbar (#8807)

* [Snackbar] Add invisible hit target for legacy snackbar

* clang

bryanoltman committed:
[Catalog] Handle dynamic type in Dragons on iOS 10 (#8827)

andrewoverton committed:
[TextControls] Use system dynamic colors for sensible defaults in iOS 13 (#8818)

MDCFilledTextField and MDCOutlinedTextField are completely unreadable (inaccessible) in iOS 13 dark mode when shown over the system background color. When using `borderStyle`, MDCBaseTextField is completely unreadable in iOS 13 dark mode when shown over _any_ color, because UITextField adds a system background color to the textfield's background. This PR addresses these issues. This PR doesn't affect pre-iOS 13 behavior.

Note that the system placeholder and clear button never pass contrast in dark mode. I know the system clear button doesn't even pass contrast in light mode.

Note that the below gifs are outdated. Here's what it looks like now in dark mode:

<img width="371" alt="Screen Shot 2019-11-14 at 11 40 07 AM" src="https://user-images.githubusercontent.com/8020010/68877158-9a036000-06d3-11ea-8cd2-902a1d2b0eb1.png">

Here's a before gif in iOS 13 dark mode:
![iOS13darkbefore](https://user-images.githubusercontent.com/8020010/68875587-15afdd80-06d1-11ea-8bee-b583a56e4537.gif)

Here's an after gif in iOS 13 light mode:
![iOS13light](https://user-images.githubusercontent.com/8020010/68875641-2c563480-06d1-11ea-8172-5743390bf839.gif)

Here's an after gif in iOS 13 dark mode:

![iOS13Dark](https://user-images.githubusercontent.com/8020010/68875625-282a1700-06d1-11ea-9f3d-2d56d068f3ed.gif)

Closes #8817.

Commit:
[AppBar] Make example titles more accurate. (#8828)

Have the titles reflect their catalog breadcrumb names.

Found while auditing dragons.

bryanoltman committed:
[Catalog] Darken ButtonBarIconExample icon colors (#8826)

* [Catalog] Darken ButtonBarIconExample icon colors

* Update BUILD file

Commit:
[Tabs] Improve custom view documentation. (#8813)

Making it clearer that types conforming to `MDCTabBarViewCustomViewable` must
always return a correct value for `intrinsicContentSize`.

Closes #8706

Commit:
merged release-candidate

Commit:
[Tabs] Mark TabBarView as accessibility tab bar. (#8811)

Marks the `MDCTabBarView` class a tab bar for UIAccessibility. This
supplements the accessibility information provided to users by including the
element's index and total number of items. In VoiceOver, this is vocalized as
something similar to "{title}, tab, 1 of 3".

Closes #8495

bryanoltman committed:
[Catalog] Fix Banner positioning in example for iOS 10 (#8829)

* [Catalog] Fix Banner positioning in example for iOS 10

* PR feedback

bryanoltman committed:
[Catalog] Remove scrim accessibility logic from BottomSheetUIControlExample (#8848)

* [BottomSheet] Ignore background tap location if VoiceOver is on

* Verify that the scrim view is focused in voice over

* cleanup

* Remove VoiceOver logic from bottom sheet, update example

* cleanup

bryanoltman committed:
[Catalog] Add VoiceOver support to BottomSheetShortCollectionExample (#8849)

Commit:
[Dragons] Treat MDCDragonsTableViewCell as an a11y button. (#8862)

The "expanded button" accessory item doesn't need to be treated as an independent button for VoiceOver because tapping the cell causes it to expand/collapse.

Fixes https://github.com/material-components/material-components-ios/pull/8856

Commit:
[Dragons] Remove unnecessary methods from MDCDragonsTableViewCell. (#8856)

Part of https://github.com/material-components/material-components-ios/issues/8852

yarneo committed:
[private] ZShadow example supports VoiceOver (#8881)

Allowing the tapable items to be correctly called out in voiceover and be interactable.

Closes #8843

Commit:
[Dragons] Simplify the MDCDragonsTableViewCell accessory types. (#8863)

The accessory views do not need to be buttons. They are now image views and the names of the properties have been updated to reflect their functional purpose. From a public point of view, the properties only need to be UIViews, so the property types have been updated accordingly as well.

Polish as part of https://github.com/material-components/material-components-ios/pull/8856

randallli committed:
[TextFields] Added tests for accessibilityLabel (#8830)

Related to https://github.com/material-components/material-components-ios/pull/8804 which got feedback asking for tests. This puts in tests before that change.

more info at #7513

Commit:
[Ink] Consolidate example code into a single file. (#8890)

Cleanup as part of https://github.com/material-components/material-components-ios/issues/8883

randallli committed:
[Typography] Using safeSharedApplication in test (#8815)

I needed to change this to run test locally in my simulator.

I'm not sure why this wasn't caught in CI.

randallli committed:
[TextFields] Added header comment: placeholder APIs are for `label text` (#8803)

Added header comment clarifying that the placeholder APIs are really for label text and not placeholder.

more info at #7513

yarneo committed:
[ActivityIndicator] Add a voiceover layout change notification… (#8909)

Adds a voiceover layout change notification and re-focus button after animation in ActivityIndicator Transition example.

Closes #8841

Commit:
[Cards] Make the Shaped Edit/Reorder example class/files match the name of the example. (#8899)

Cleanup as part of https://github.com/material-components/material-components-ios/issues/8866

Commit:
[Ink] Make the ink views accessibility elements. (#8891)

Stacked on top of https://github.com/material-components/material-components-ios/pull/8890

This allows the ink views to be focused by VoiceOver and tapped as buttons to initiate the ink effect.

Closes https://github.com/material-components/material-components-ios/issues/8883

Commit:
[Tabs] Use built-in "tabs" behavior on iOS 10+ (#8812)

Although not available on iOS 9, on iOS 10 the MDCTabBar can use the
UIAccessibilityTraitTabBar so that its accessibility elements will be
decorated using the "tab, 1 of 3" information automatically. This change also
improves interaction with Voice Control because that relies on matching the
`accessibilityLabel`.

Improves #6757 for iOS 10+

andrewoverton committed:
[TextControls] Add dynamic type functionality to text controls example (#8847)

This PR adds some dynamic type functionality to the text controls example.
Note that in the largest content sizes the content goes off the screen. This will be addressed in a follow up PR.
 
Here is a gif:

![DynamicType](https://user-images.githubusercontent.com/8020010/68900027-f92b9980-0700-11ea-958e-6965638bdbc5.gif)

Closes #8845.

randallli committed:
Fix spelling mistake (#8910)

spelling mistake was made in https://github.com/material-components/material-components-ios/pull/8830

while working on #7513

yarneo committed:
[Buttons] Fix a11y issues in Buttons (Content Edge Insets) exam… (#8900)

In this PR we are fixing the non-accessible colors that were caused due to the container scheme being nil, as init isn't being called with a Storyboard, but rather initWithCode.

Secondly, now the FAB has a good a11y label, so it can be presented correctly when focused.

Before:
![Simulator Screen Shot - iPhone 7 - 2019-11-15 at 09 53 25](https://user-images.githubusercontent.com/4066863/68952401-c8de0c80-078d-11ea-8816-d1e667fbef6b.png)

After:
![Simulator Screen Shot - iPhone 7 - 2019-11-15 at 09 52 50](https://user-images.githubusercontent.com/4066863/68952352-b4017900-078d-11ea-8cf9-08c081a779d9.png)

Closes #8831

yarneo committed:
[PageControl] Fixes contrast ratio and button interactivity for… (#8922)

PageControlAnimationBlockExample has too-low contrast in the controls. It also does not disable the +2/-2 buttons even when they have no effect.

Updated colors to have an accessible contrast ratio, and updated the states of the buttons to show correct interactivity (enabled/disabled)

![Simulator Screen Shot - iPhone 7 - 2019-11-15 at 10 54 32](https://user-images.githubusercontent.com/4066863/68956562-532a6e80-0796-11ea-9378-463988188a47.png)

Closes #8919

yarneo committed:
[Ripple] Ripple typical example now can be used with Voice Over (#8924)

All views in example are accessible with Voice Over and can be activated.
All views have a relevant accessibility label and the appropriate traits.

Closes #8913

Commit:
[Cards] Make the shaped collection cell an accessibility button. (#8901)

Fixes https://github.com/material-components/material-components-ios/issues/8866

codeman7 committed:
[Buttons] Update accessibility label for FABs (#8911)

This change updates the accessibility label for both FABs within the `Buttons (Swift and Storyboard)` example.

Closes #8854

bryanoltman committed:
[Catalog] Add accessibility elements to BottomSheetExamples (#8923)

* [Catalog] Add accessibility element to BottomSheetTallExample

* [Catalog] Add accessibility element to BottomSheetSimpleExample

* Add dismissOnBackgroundTap to BottomSheetPresentationExample

* clang

Commit:
[Tabs] Add newlines after all Swift doc headers. (#8931)

Fixes https://github.com/material-components/material-components-ios/issues/8427

yarneo committed:
make card with ripple example a5able (#8929)

Voice Over and Switch Control users can activate the Card to show Ripple.
The card has an accessibility label and an accessibility trait to indicate that it is interactive.

Closes #8914

codeman7 committed:
[BottomNavigation] Clean up badge value test example (#8921)

This cleans up the bottom navigation `Badge Value Test` example to have the correct imports, use our modern theming, and removes semicolons.

## Screenshots

| Before | After |
| --- | --- | 
|![Simulator Screen Shot - iPhone 11 Pro Max - 2019-11-15 at 08 29 50](https://user-images.githubusercontent.com/7131294/68959206-7a774080-0782-11ea-9edf-6bab766b3c4c.png)|![Simulator Screen Shot - iPhone 11 Pro Max - 2019-11-15 at 08 31 40](https://user-images.githubusercontent.com/7131294/68959214-7fd48b00-0782-11ea-84ec-0b6087e40bfd.png)|



NO_BUG=This is cleanup I found while working on #8837

randallli committed:
[TextFields] Check if placeholder is hidden before verbalizing it in voice over. (#8804)

[TextFields] Add check for placeholder before adding to the accessibility strings spoken by voice over.

Also added changed example to remove the floating label so that we can check that voice over correctly verbalizes only the visible text.

Before and After zip:
[BeforeAfterVideos.zip](https://github.com/material-components/material-components-ios/files/3843558/BeforeAfterVideos.zip)

QA=Notice that "state" is verbalized when it is not visible on screen in the after.
more info at #7513

yarneo committed:
adds a11y support for page control with next button example (#8932)

PageControlAnimationBlockExample has too-low contrast in the controls. It also does not disable the next button even when it has no effect.

Updated colors to have an accessible contrast ratio, and updated the state of the button to show correct interactivity (enabled/disabled)

Before:
![Simulator Screen Shot - iPhone 7 - 2019-11-15 at 12 12 34](https://user-images.githubusercontent.com/4066863/68961835-3cd5e000-07a1-11ea-99fd-83fd7a1e481b.png)


After:
![Simulator Screen Shot - iPhone 7 - 2019-11-15 at 11 40 59](https://user-images.githubusercontent.com/4066863/68959927-1ada5e80-079d-11ea-9b84-7e8d2cfe1ff7.png)


Closes #8904

bryanoltman committed:
[Catalog] Improve contrast ratio in PageControlSwiftExampleViewController (#8938)

Commit:
[Dialogs] Make the "Dialogs with accessory view" example's cells accessibility buttons. (#8939)

Fixes https://github.com/material-components/material-components-ios/issues/8880

bryanoltman committed:
[Catalog] Improve contrast ratio in PageControlTypicalUseViewController (#8941)

randallli committed:
[MDCTextfields] Added accessibility label tests for MDCMultilineTextFields (#8902)

Copied the logic from the single line textfield. over to the multiline one.

part of https://github.com/material-components/material-components-ios/issues/7513

Commit:
[ShadowElevations] Provide semantic VoiceOver labels when interacting with the typical use example. (#8930)

Before this change, VoiceOver would announce the slider's percentage value as the value changed up and down.

After this change, VoiceOver will announce the select shadow elevation name, if available, otherwise it will announce a fallback string with the selected shadow elevation.

Fixes #8927

bryanoltman committed:
[Catalog] Fix accessibility labels and layout in CollectionListCellExampleTypicalUse (#8944)

* [Catalog] Fix accessibility labels and layout in CollectionListCellExampleTypicalUse

* clang

Commit:
[Cards] Make the "Typical use" example title label an a11y heading. (#8945)

Closes https://github.com/material-components/material-components-ios/issues/8861

Commit:
[Chips] Fix layout of ChipsSizingExampleViewController. (#8940)

Part of https://github.com/material-components/material-components-ios/issues/8868

| Before | After |
|:--|:--|
| ![Simulator Screen Shot - iPhone Xʀ - 2019-11-15 at 12 20 17](https://user-images.githubusercontent.com/45670/68962319-4c095d80-07a2-11ea-861c-04244f213da1.png) | ![Simulator Screen Shot - iPhone Xʀ - 2019-11-15 at 12 19 39](https://user-images.githubusercontent.com/45670/68962302-40b63200-07a2-11ea-918e-b62c151a176d.png) |

codeman7 committed:
[BottomNavigation] Add accessibility value for badge (#8920)

This updates the `Badge Value Test` example to have an accessibility label for the tab with a badge.

Closes #8837

bryanoltman committed:
[Catalog] Delete non-functional ShadowElevationsTypicalUseViewController.swift (#8955)

Commit:
[Dialogs] Make the "View with corner radius" example's dialog escapable in VoiceOver. (#8950)

This PR makes the dialog an accessibility button with a hint that tapping the dialog in VoiceOver will dismiss the dialog.

Fixes #8876

randallli committed:
[TextFields] simplify test names since the name of the file already gives context. (#8952)

this was feed back in https://github.com/material-components/material-components-ios/pull/8902#discussion_r346874329

Applying change here.

bryanoltman committed:
[Catalog] Add close button to bottom drawer example header (#8963)

* [Catalog] Add close button to bottom drawer example header

* formatting

* Add heading accessibility trait to DrawerHeaderViewController titleLabel

yarneo committed:
[FeatureHighlight] Add a voiceover dismissal affordance for the feature highlight. (#8959)

Currently the feature highlight didn't have a proper dismiss affordance when voiceover is on.

It will instead speak a hint of "double tap to dismiss" on the title but without allowing the user to dismiss even when double tapped.

This adds a dismiss affordance and passes the accessibility hint to dismiss to that view.

Frame of dismissal button in VO:
![IMG_2140](https://user-images.githubusercontent.com/4066863/68971016-95fc3e80-07b6-11ea-846d-e4219429a4d7.PNG)


Closes #8961 #8888 #8884 #8886 #8885

Commit:
[Dialogs] Make all "Dismissing Dialogs" cells a11y buttons. (#8956)

Part of https://github.com/material-components/material-components-ios/issues/8878

Commit:
[Buttons] Improve accessibility of Shaped Buttons. (#8964)

Improve the Shaped Buttons example accessibility by providing better name to
the buttons and setting an accessibility label on the FAB.

|Before|After|
|---|---|
|![IMG_0105](https://user-images.githubusercontent.com/1753199/68969207-6b0feb80-07b2-11ea-96df-69475c4b3477.PNG)|![IMG_0106](https://user-images.githubusercontent.com/1753199/68969208-6d724580-07b2-11ea-8fe2-aa7c01082496.PNG)|



Closes #8853

Commit:
[Dialogs] More Material Alerts examples is more accessible. (#8960)

Provides accessibility traits to the cells and marks the cells as
accessibility elements.

|Before|After|
|---|---|
|![IMG_0104](https://user-images.githubusercontent.com/1753199/68968707-40716300-07b1-11ea-9227-539c7c6dfc3a.PNG)|![IMG_0103](https://user-images.githubusercontent.com/1753199/68968677-2cc5fc80-07b1-11ea-88db-1c0f5bb14652.PNG)|


Closes #8877

Commit:
[Dialogs] Provide accessibility traits for Input Field example (#8958)

Set the correct accessibility traits for the collection cell that presents a
Dialog.

|Before|After|
|---|---|
|![IMG_0065](https://user-images.githubusercontent.com/1753199/68968199-02c00a80-07b0-11ea-9cc4-59b98e31e4ba.PNG)|![IMG_0102](https://user-images.githubusercontent.com/1753199/68968219-0b184580-07b0-11ea-8c70-4273603e88e7.PNG)|



Closes #8879

Commit:
[FeatureHighlight] Make "Color" example accessible. (#8957)

Includes the following changes:
1.  Each cell is an accessibility element with the color name as the label.
2.  Shown Feature Highlights no longer include the "double-tap to dismiss"
    hint.

|Before|After|
|---|---|
|![IMG_0099](https://user-images.githubusercontent.com/1753199/68967680-c344ee80-07ae-11ea-8b97-43a0045ffc53.PNG)|![IMG_0098](https://user-images.githubusercontent.com/1753199/68967657-b4f6d280-07ae-11ea-828f-4976f05ce862.PNG)|


Closes #8889

codeman7 committed:
[ActionSheet] Allow Swift example to support dynamic type (#8946)

Perviously the ActionSheet swift example did not support dynamic type. This updates the example table to support dynamic type.

## Screenshots

| Before | After |
|---|---|
|![Simulator Screen Shot - iPhone 5 - 2019-11-15 at 10 11 58](https://user-images.githubusercontent.com/7131294/68965314-5f133200-0790-11ea-988f-63871d8f5183.png)|![Simulator Screen Shot - iPhone 5 - 2019-11-15 at 10 11 20](https://user-images.githubusercontent.com/7131294/68965322-63d7e600-0790-11ea-826c-a7724b8935c5.png)|



Closes #8820

galiak11 committed:
[MDC Dragons] Fix MDCBaseCell Example to work with Voice Over (#8953)

Make sure List cells are accessible to Voice Over and have a label that distinguish them from each other, and have the selected trait when selected.

Commit:
[Chips] Give the ChipsSizing example's sliders a11y labels and values. (#8942)

Stacked on https://github.com/material-components/material-components-ios/pull/8940

Part of https://github.com/material-components/material-components-ios/issues/8868

yarneo committed:
[FeatureHighlight] Positions label not hidden behind nav bar. (#8969)

Now label can be seen and voiceover will focus on it in the right order.

Closes #8888 

Before:

![Simulator Screen Shot - iPhone 7 - 2019-11-15 at 14 53 43](https://user-images.githubusercontent.com/4066863/68971687-18393280-07b8-11ea-9b1c-b8139f2946b2.png)


After:

![Simulator Screen Shot - iPhone 7 - 2019-11-15 at 14 53 34](https://user-images.githubusercontent.com/4066863/68971698-1b342300-07b8-11ea-8bca-962f50a31e6d.png)

Commit:
[List] Make Base Cell example more accessible. (#8947)

Allows for interaction using Voice Over and Switch Control.

|Before|After|
|---|---|
|![IMG_0100](https://user-images.githubusercontent.com/1753199/68967740-e96a8e80-07ae-11ea-8309-7abb3baa9b39.PNG)|![IMG_0101](https://user-images.githubusercontent.com/1753199/68967753-f5565080-07ae-11ea-88bc-bb18e1ef0bcf.PNG)|



Part of #8898

Commit:
[Dialogs] Implement the a11y escape gesture on all custom dialogs in the "Dismissing Dialogs" example. (#8965)

Part of https://github.com/material-components/material-components-ios/issues/8878

galiak11 committed:
[MDC Dragons] AppBar example has low-contrast text (#8966)

Replacing old "Themers" in "Presented" AppBar example with new theming extensions - in order to improve contrast ratio.

codeman7 committed:
[Shadow] Make Shadow Corner Radius example more accessible. (#8962)

This updates the Shadow Corner Radius example to be more accessible by adding accessibility label and hint to the slider in this example.

Closes #8928

Commit:
[Cards] Fix bug in the "Typical use" example where the image would not appear. (#8943)

The image was not appearing because the card using a custom image view that was trying to inherit the mask of the card's shape layer.

This change removes the custom image view and instead applies custom corner radii to the image view directly. This is not ideal, but is an improvement. Ideally, the card would provide a content view that is clipped to the card's shape. It's not possible to enable clipping on the card itself because this causes the shadow to go away.

| Before | After |
|:--|:--|
| ![Simulator Screen Shot - iPhone Xʀ - 2019-11-15 at 12 57 37](https://user-images.githubusercontent.com/45670/68964703-08b1ed80-07a8-11ea-978c-2c71145ca8c3.png) | ![Simulator Screen Shot - iPhone Xʀ - 2019-11-15 at 12 56 52](https://user-images.githubusercontent.com/45670/68964693-02bc0c80-07a8-11ea-9d3d-644b81aebee6.png) |

Part of https://github.com/material-components/material-components-ios/issues/8861

Commit:
[Dialogs] Fix bug in "Dismissing Dialogs" dialogs where the buttons were not themed. (#8967)

The dialog view controllers were not being provided with the container scheme.

Fixes https://github.com/material-components/material-components-ios/issues/8878

| Before | After |
|:--|:--|
| ![Simulator Screen Shot - iPhone Xʀ - 2019-11-15 at 14 41 13](https://user-images.githubusercontent.com/45670/68970747-fe96eb80-07b5-11ea-8c69-8e0df6ffd64e.png) | ![Simulator Screen Shot - iPhone Xʀ - 2019-11-15 at 14 40 46](https://user-images.githubusercontent.com/45670/68970733-f5a61a00-07b5-11ea-81c4-b7fa75a29f84.png) |

fluxminds committed:
[Banner] Add a snapshot for banner with no action case. (#8968)

Pre-work of fixing https://github.com/material-components/material-components-ios/issues/8954..

codeman7 committed:
[List] Fix catalog compile issue (#8974)

This fixes a compile issue within the catalog where we were accidentally formatting a string wrong.

NO_BUG=Fast follow on #8953

fluxminds committed:
[Banner] Support use case where no button appears on Banner. (#8951)

closes https://github.com/material-components/material-components-ios/issues/8954.

When all buttons are hidden on the banner, we don't want to let the hidden button to occupy an invisible area which affects textView's layout.

randallli committed:
[TextFields] Correction to the accessibility label (#8976)

UITextField verbalizes the placeholder in a lower pitch when no text has been entered. This change incorporates this behavior into what values get added to the MDCTextField's accessibilityLabel.

Before after videos:
[singleFix.zip](https://github.com/material-components/material-components-ios/files/3853326/singleFix.zip)


QA=Notice that "state" is verbalized when it is not visible on screen in the after.
more info at #7513

Commit:
[ListItems] Make Base Cell example more accessible. (#8982)

Provides accessibility labels and distinguishing colors for each cell. Voice
Over users can now scroll the example.

Closes #8898

Commit:
[Tabs] Improve accessibility for TabBar example. (#8983)

The Tab Bar View Controller with Interface Builder example had no way to
dismiss with Voice Over, disappearing labels, and too much code.  Provides an
AppBar with back button, labels on every child view controller, and better
support for Dynamic Type.

|Before|After 1 AXXXL|After 2 XS|
|---|---|---|
|![IMG_0110](https://user-images.githubusercontent.com/1753199/68987915-134aa200-07fd-11ea-8424-277e0b6eb1ee.PNG)|![IMG_0114](https://user-images.githubusercontent.com/1753199/68987917-1d6ca080-07fd-11ea-8a1c-d1c372665fe5.PNG)|![IMG_0115](https://user-images.githubusercontent.com/1753199/68987918-22c9eb00-07fd-11ea-8b74-ec85cb602e93.PNG)|

Closes #8978

fluxminds committed:
[TextFields] Add missing Button dependency. (#8995)

closes https://github.com/material-components/material-components-ios/issues/8996.

Fix import dependency for https://github.com/material-components/material-components-ios/blob/develop/components/TextFields/src/private/MDCTextInputCommonFundament.m#L17.

Commit:
Automatic changelog preparation for release.

andrewoverton committed:
Hand-modified CHANGELOG.md API diff.

Commit:
Hand-modified CHANGELOG.md API diff.

fluxminds committed:
[ShadowElevation] Delete swift example target since there is no input file. (#9005)

Follow up PR to https://github.com/material-components/material-components-ios/pull/8955.

Commit:
Hand-modified CHANGELOG.md API diff.

romoore commented on CHANGELOG.md:
```diff
@@ -1,3 +1,191 @@
+### Slider
+
+* [Update documentation to match behavior. (#8782)](https://github.com/material-components/material-components-ios/commit/3671e05bde32bf2bfb693d3eb6e21dcf87abe4e8) (Robert Moore)
+* [Fix event handling. (#8759)](https://github.com/material-components/material-components-ios/commit/4194c01fbe880d02ad6798c5a02ea471e37da0be) (Robert Moore)
```
I believe this was included in 94.2.0 and should be removed.

fluxminds replied:
Addressed in https://github.com/material-components/material-components-ios/pull/8994/commits/eff2b7b1f60ac7a72b39ea27c4dd7ccaf594f7f4.

romoore reviewed (changes requested):
Please bump the version number.

Commit:
[FeatureHighlight] Fix accessibility tests. (#9018)

Pull request #8959 changed the accessibility behavior of Feature Highlight but
did not update the tests. Somehow the last commit in the PR failed CI tests,
but the PR itself was marked as passing.  This commit updates the unit tests
to reflect the latest behavioral expectations for accessibility.

Follow-up for #8961

Commit:
Revert "[TextFields] Correction to the accessibility label (#8976)"

This reverts commit c498516b358998c9853e9934df9b643d4256b9da.

Commit:
Revert "[TextFields] simplify test names since the name of the file already gives context. (#8952)"

This reverts commit d6ae7f6782bb554d043879b5965e9b2708145a60.

Commit:
Revert "[TextFields] Check if placeholder is hidden before verbalizing it in voice over. (#8804)"

This reverts commit f9c17caf1bffd8e4538ee2c587d780e55437243c.

Commit:
Hand-modified CHANGELOG.md API diff.

Commit:
Hand-modified CHANGELOG.md API diff.

Commit:
Bumped version number to 94.3.0.

fluxminds commented:
Version bumped. It was planned.
